### ISSUE TITLE
Issues with macOS build

### ISSUE THREAD

**Issue by y-mx-b** [Librum-Reader/Librum#6]
When attempting a macOS build of this project, I noticed that both Boost (which I happened to have on my system) and [Boost::ext].DI (which I did not have on my system) are required by Librum. I'm not sure what version of each is required, but I do think both should be appended to the list of required build dependencies.

I did manage to successfully complete a macOS build after just copy-pasting the `di.hpp` file and changing the include statements to be absolute, though. I haven't tested the application out yet, so I don't know how functional this build is.

**Comment by denniskaydalov** [comment]
[Boost::ext].DI was also a dependency for my Linux system, I solved the problem the same way by copy and pasting. Either some code that automatically installs this dependency is broken, or [[Boost::ext].DI ](https://github.com/boost-ext/di)should be added to the list of dependencies.

**Comment by Creapermann** [comment]
Hey, I'll add the dependencies to the setup guide, thanks for mentioning. Note, Mac OS is not officially supported yet, I didnt have a chance to test it there so I dont know how the application looks, or if it even runs. 
It'd be great if you could give me some feedback on how it went

**Comment by Creapermann** [comment]
It should work with the newest commit (0d3799efe93a1a0929350710daf0a5bd7510ae62)

**Comment by y-mx-b** [comment]
> It should work with the newest commit ([0d3799e](https://github.com/Librum-Reader/Librum/commit/0d3799efe93a1a0929350710daf0a5bd7510ae62))

Thanks, I've managed a successful build now. I'll try registering a new account and checking the functionality of the app later. I can confirm that the app launches (much faster than it did the first time, for some reason?) and that the login and registration screens both work. I am getting an error message in my terminal after launching it though.

```
qrc:/qt-project.org/imports/QtQuick/Controls/macOS/TextField.qml:46: TypeError: Property 'styleFont' of object QQuickRectangle(0x6000012fe680) is not a function
```

Not sure if this is known or not.

**Comment by Creapermann** [comment]
Even tho I've never tested it on a mac, I know that this would happen on mac. It is connected to the way I load the fonts into the application, afaik. this needs to be treated specially on mac. 
I won't be able to fix this any soon since I don't yet have access to a mac. When I'll list mac as officially supported, this should be patched.

You could also try and fix it yourself and send a PR. The important parts would be in src/CMakeLists.txt on line 78 and in the src/main.cpp on line 76 (loadFonts())

**Comment by Creapermann** [comment]
Also, you don't need to register an account (even tho you could by running the Librum server in the background locally). If you leave the email and password empty, you should be redirected to the home screen.

**Comment by y-mx-b** [comment]
> Also, you don't need to register an account (even tho you could by running the Librum server in the background locally). If you leave the email and password empty, you should be redirected to the home screen.

Alright, I've taken a quick look around the macOS build I have locally, and a lot of things seem to be broken. The downloads page appears as shown below, with all the covers and authors being the same and the title shown on the previews being broken. None of the search filters seem to do anything. Attempting to download a book does nothing. 

![Screen Shot 2022-08-13 at 4 56 08 PM](https://user-images.githubusercontent.com/48655352/184474800-8e3bb822-f14e-4cc5-910a-fe2a97d456fa.png)

![Screen Shot 2022-08-13 at 5 02 37 PM](https://user-images.githubusercontent.com/48655352/184474981-b1160bb8-7812-4948-8082-ba1a53ac6593.png)

The home page looks as shown below, and attempting to add a book I have downloaded locally bugs the page out. Similar to the downloads page, none of the filters seem to do anything. It goes back to normal (an empty home page) after clicking into a different page and going back (no books have been added).

![Screen Shot 2022-08-13 at 4 57 36 PM](https://user-images.githubusercontent.com/48655352/184474837-4505b392-75d6-4018-ba89-d8926ebaf76b.png)

![Screen Shot 2022-08-13 at 4 58 43 PM](https://user-images.githubusercontent.com/48655352/184474872-79accd5a-1fb9-4e33-8e3d-f536519837ec.png)

The statistics page and add-ons page seem to be just placeholders. I'm not aware of whether this is intentional or not (the readme doesn't indicate that these features aren't implemented yet).

![Screen Shot 2022-08-13 at 5 01 37 PM](https://user-images.githubusercontent.com/48655352/184474955-0270b9ae-f454-40e9-bef8-d8d6fe89b2f9.png)

![Screen Shot 2022-08-13 at 5 01 58 PM](https://user-images.githubusercontent.com/48655352/184474961-409764a4-a85e-4e84-aaae-46ee33dde6bc.png)

The settings page doesn't seem broken, although attempting to change the appearance to dark mode doesn't do anything. None of the settings are permanent though, they will reset upon closing and reopening the app. The storage page also seems a little off. It claims I have 24 files stored (I haven't downloaded anything successfully, and attempting more downloads doesn't change this number; is this a placeholder for if you haven't created an account?). There are also a couple grammatical errors on some of the settings pages ("This are" instead of "This is" and a missing comma before the "click here" link on this page, "Any new update?" instead of "Any new updates?" under the "Updates" page, etc.).

![Screen Shot 2022-08-13 at 5 06 28 PM](https://user-images.githubusercontent.com/48655352/184475087-b781c32c-1a79-48ce-8d41-70d6d7147dff.png)

**Comment by Creapermann** [comment]
Well, it builds, which is good. Librum is still in an early stage of development, many functionalities are not yet buid in or ready. The app is not ready to use yet, even tho the server side is functional, I still need to connect many functionalities from the clientside. 
The placeholders you found are placed there on purpose (I didnt really grammar-prove them since they are place holders, I'll patch this soon).

**Comment by Creapermann** [comment]
I see how the application not yet being functional isn't obvious by looking at the README file, I fixed this with the last commit (5390e2692c4b6211199e4eb965f3770c6be3612a)
